src/telemetry/logging.rs

```diff
@@ -66,15 +71,29 @@ pub fn build_subscriber(verbose: bool, appender: Option<RollingFileAppender>) ->
         // Force the file logger to log at `debug` level
         let file_env_filter = EnvFilter::from("magi=debug,network=debug");
 
-        tracing_subscriber::registry()
-            .with(stdout_formatting_layer)
-            .with(
-                tracing_subscriber::fmt::layer()
-                    .with_ansi(false)
-                    .with_writer(non_blocking)
-                    .with_filter(file_env_filter),
-            )
-            .init();
+        if json_logs {
+            tracing_subscriber::registry()
+                .with(tracing_subscriber::fmt::layer().json())
+                .with(
+                    tracing_subscriber::fmt::layer()
+                        .with_ansi(false)
+                        .with_writer(non_blocking)
+                        .with_filter(file_env_filter),
+                )
+                .init();
+        } else {
+            tracing_subscriber::registry()
+                .with(stdout_formatting_layer)
+                .with(
+                    tracing_subscriber::fmt::layer()
+                        .with_ansi(false)
+                        .with_writer(non_blocking)
+                        .with_filter(file_env_filter),
+                )
+                .init();
+        }
+    } else if json_logs {
+        tracing_subscriber::fmt().json().init();
```

Comment:
feel like this could be rewritten to take out some branching but havent given it much thought